tls/s2n_x509_validator.c

```diff
@@ -436,26 +436,59 @@ S2N_RESULT s2n_x509_validator_check_cert_preferences(struct s2n_connection *conn
      *# "signature_algorithms" extension also applies to signatures appearing in
      *# certificates.
      */
-    if (security_policy->certificate_signature_preferences == NULL) {
+    if (security_policy->certificate_signature_preferences == NULL
+            && security_policy->certificate_key_preferences == NULL) {
         return S2N_RESULT_OK;
     }
```

Comment:
If you code can handle either of these being NULL anyway (since you only bail if both are NULL), why add the complication of bailing early?